Argo/Functions/decode.swift

```diff
@@ -1,35 +1,215 @@
-// MARK: decode root objects and arrays
+/**
+  Decode a raw JSON value into an optional `Decodable` value
 
+  This function takes the output from `NSJSONSerialization` and attempts to
+  decode it into a `Decodable` value, returning an `Optional`. This works based
+  on the type you ask for.
+
+  For example, the following code attempts to decode an optional `String`,
+  because that's what we have explicitly stated is the return type:
+
+  ```
+  do {
+    let object = try NSJSONSerialization.JSONObjectWithData(data, options: nil)
+    let str: String? = decode(object)
+  } catch {
+    // handle error
+  }
+  ```
+
+  - parameter object: The output from `NSJSONSerialization`
+  - returns: An `Optional` value of type `T` where `T` is `Decodable`
+*/
 public func decode<T: Decodable where T == T.DecodedType>(object: AnyObject) -> T? {
   return decode(object).value
 }
 
+/**
+  Decode a raw JSON value into an optional array of `Decodable` values
```

Comment:
@gfontenot do you think this is a valuable place to have that discussion? I think getting language we're happy with in the documentation is a good start.

How do you feel about "Transforms a raw JSON value into an optional array of Decodable values" instead of "Parses"? That way it's about data type transformation, and we leave the loaded words parse/decode out of it?